lib/elixir/test/elixir/file_test.exs

```diff
@@ -1322,6 +1322,26 @@ defmodule FileTest do
     assert Enum.count(stream) == 2
   end
 
+  test "stream keeps BOM" do
+    src = fixture_path("utf8_bom.txt")
+    bom_line = src
+    |> File.stream!()
+    |> Enum.take(1)
+    |> List.first()
+
+    assert <<239, 187, 191>> <> "Русский\n" = bom_line
```

Comment:
I think we can use `==` operator in assertion, it gives better message on failure than with `=`.